testing/log/service.go

```diff
@@ -183,18 +184,18 @@ func (s *service) tryReadSnapshot(service storage.Service, object storage.Object
 		if err != nil {
 			return nil, err
 		}
-		if recentSnapshot, err := service.StorageObject(object.URL()); err == nil {
-			if recentSnapshot.FileInfo().Size() == fileSize { //no content modification since last content download, it safe to return snapshot
+		if recentSnapshot, err := fs.Object(context.Background(), object.URL()); err == nil {
```

Comment:
you can use err from `reader, err := fs.Download(context.Background(), object)` instead of creating new one.